core/src/main/java/org/apache/iceberg/avro/GenericAvroWriter.java

```diff
@@ -133,4 +141,38 @@ private WriteBuilder() {
       }
     }
   }
+
+  public static class UnionSchemaWriter<V extends Object> implements ValueWriter<V> {
```

Comment:
Do we need to support writing out union types? Iceberg types does not support union and almost all compute engines do not support unions as well. I'm not sure if there's a usecase here to support it.

On the read side it makes sense to us to support reading unions, because that data could have been written by non Iceberg writers.

What are your thoughts?

Reply:
correct, 
since iceberg does not support union type but avro does, this approach is similar to spark-avro where union types are converted to record.
I dont think it make sense to support union types as it will mean adding support in schema for that which is larger change and has more blast radius.
spark does not support union types so it make sense to align with spark and keep scope limited IMO

Reply:
Files written by Iceberg must always conform to the Iceberg spec. Iceberg tables can allow reading data that is imported, but should _never_ write data that cannot be read by a generic implementation of the Iceberg spec.